csrc/counter_kdf.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+#include "buffer.h"
+#include "env.h"
+#include "generated-headers.h"
+#include <openssl/evp.h>
+#include <openssl/kdf.h>
+
+using namespace AmazonCorrettoCryptoProvider;
+
+extern "C" JNIEXPORT void Java_com_amazon_corretto_crypto_provider_CounterKdfSpi_nKdf(JNIEnv* env,
+    jclass,
+    jint digestCode,
+    jbyteArray jSecret,
+    jint secretLen,
+    jbyteArray jInfo,
+    jint infoLen,
+    jbyteArray jOutput,
+    jint outputLen)
+{
+    try {
+        EVP_MD const* digest = digest_code_to_EVP_MD(digestCode);
```

Comment:
`digest_code_to_EVP_MD` [won't throw](https://github.com/corretto/amazon-corretto-crypto-provider/blob/cbabe7a56af5a657187b871ae17948724827a1e5/csrc/util.cpp#L55) for SHA1, but we don't support SHA1 Counter KDF. how should we handle this?

Reply:
This case is handled on the Java side. We only instantiate the Spi class for the desired digests. SHA1 is still allowed with HKDF and `digest_code_to_EVP_MD` method is used for HKDF too.